src/get-started/java-binding.md

```diff
@@ -880,56 +873,319 @@ Currently Java Binding includes the following built-in services:
+- `--node-config` for a path to the final node configuration file.
+- `--artifacts-path` for a path to the directory containing compiled service
+  artifacts.
+- `--ejb-port` for port that Java services will use for communication.
+  Java Binding does not use public API address directly.
+- `--public-api-address` and `--private-api-address` for socket addresses for
+  node user API. Public API address is used by users to send transactions and
+  requests to the blockchain, private one is used by node administrators to
+  perform different maintenance actions. Setting these addresses isn't strictly
+  necessary for every node, but we will need available API ports for deploying
+  and starting service using `exonum-launcher`.
+
+There are also optional parameters useful for debugging purposes, logging
+configuration and JVM fine tuning:
+
+- `--jvm-args-prepend` and `--jvm-args-append`: Additional parameters for JVM
+  that prepend and append the rest of arguments. Must not have a leading dash.
+  For example, `Xmx2G`.
+- `--jvm-debug`: Allows JVM being remotely debugged over the JDWP protocol.
+  Takes a socket address as a parameter in form of `HOSTNAME:PORT`. For example,
+  `localhost:8000`.
+- `--ejb-log-config-path` for path to Log4j 2 configuration file. Default config
+  `log4j-fallback.xml` provided with Exonum Java app prints to STDOUT.
+ 
+```$sh
+$ exonum-java run \
+    --db-path testnet/db \
+    --node-config testnet/node.toml \
+    --artifacts-path artifacts \
+    --ejb-port 7000 \
+    --ejb-log-config-path "log4j.xml" \
+    --master-key-pass pass \
+    --public-api-address 127.0.0.1:3000 \
+    --private-api-address 127.0.0.1:3010
 ```
 
+#### Changing the used JVM
+
+By default, Exonum Java App automatically finds system JVM and uses it. To
+change this, the user can set `JAVA_HOME` environment variable pointing at the
+JVM installation directory.
+
+#### Debugging the JVM
+
+To enable remote debugging of Java code on a running Exonum node, 
+pass `--jvm-debug` option with a socket address to connect to
+from a debugger:
+
+```sh
+$ exonum-java run -d testnet/db -c testnet/node.toml \
+    --public-api-address 127.0.0.1:3000 \
+    --private-api-address 127.0.0.1:3010 \
+    --ejb-port 7000 \
+    --jvm-debug localhost:8000
+```
+
+Now you can debug the service using any JDWP client, such as command line
+JDB or a debugger built in your IDE:
+
+```sh
+$ jdb -attach localhost:8000 -sourcepath /path/to/source
+```
+
+## Deploy and Start Service
+
+Exonum Java Binding provides a way to dynamically deploy and start multiple
+services without stopping the nodes in the network. This can be done by sending
+particular transactions to the built-in Supervisor service. To simplify this
+process for the users, `exonum-launcher` is recommended to be used.
+
+Compiled services JAR files must be placed into the directory `artifacts-path`
+argument points to. Service files are needed for the whole time of their
```

Comment:
Is this true?

Reply:
Yes, they must not be moved or modified once deployed.